src/main/java/ObjectPool.java

```diff
@@ -0,0 +1,81 @@
+import vi.viredis.client.ViRedis;
+
+import java.io.IOException;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedTransferQueue;
+
+public abstract class ObjectPool<T> implements Pool<T> {
+
+    private int size;
+
+    private boolean shutdown;
+
+    private String host;
+    private int port;
+
+    private BlockingQueue objects;
+
+    public ObjectPool(int size, String host, int port) {
+        this.size = size;
+        this.host = host;
+        this.port = port;
+        shutdown = false;
+        init();
+    }
+
+    /*
+     * initiate the pool with fix size
+     */
+    private void init() {
+        objects = new LinkedTransferQueue();
+        for (int i = 0; i < size; i++) {
+            objects.add(createNew(host, port));
+        }
+    }
+
+    private T createNew(String host, int port) {
+        try {
+            return (T) new ViRedis(host, port);
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    @Override
+    public ViRedis get() {
+        if (!shutdown) {
+            ViRedis t = null;
+
+            try {
+                t = (ViRedis)objects.take();
+            }
+            catch (Exception e) {
+                e.printStackTrace();
+            }
+
+            return t;
+        }
+
+        throw new IllegalStateException("Object pool is already shutdown.");
+    }
+
+    @Override
+    public void release(T t) {
+        try {
+            objects.offer(t);
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Never print to stacktrace. Throw an exception or handle the exception.

Reply:
done!